### PR TITLE
EPMRPP-60608 - add description suite/step and unit tests

### PR DESCRIPTION
**Pull request by chivekrodis**

### CODE REVIEW

File: src/__tests__/reporter/reporter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+// @ts-ignore
+import RPClient from '@reportportal/client-javascript';
```
Comment (KatsiarynaTatarynovich) [inline src/__tests__/reporter/reporter.spec.ts]: Why did you add `RPClient` if you do not use it?

File: src/__tests__/reporter/reporter.spec.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import { Reporter } from '../../reporter';
+import { getDefaultMockConfig, RPClientMock } from './RPClientMock';
+import { StartLaunchRQ, StartTestItemRQ } from '../../models';
+import { STATUSES, TEST_ITEM_TYPES } from '../../constants';
+import { getStartLaunchObj } from '../../utils';
+
+const setupReporter = (fields: Array<string> = []): Reporter => {
+  const config = getDefaultMockConfig();
```
Comment (KatsiarynaTatarynovich) [inline src/__tests__/reporter/reporter.spec.ts]: You use the same function as on line 43. It will be better to move `const config = getDefaultMockConfig ();` before `setupReporter` and use it inside of it.

File: src/__tests__/reporter/reporter.spec.ts [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  describe('start report launch', () => {
+    const reporter = setupReporter();
+    const startTime = Date.now();
+    const startLaunchObj: StartLaunchRQ = getStartLaunchObj({ startTime }, config);
+    reporter.reportTaskStart(startTime, undefined, undefined);
+    test('client.startLaunch should be called with corresponding params', () => {
+      expect(reporter['client'].startLaunch).toHaveBeenCalledTimes(1);
+      expect(reporter['client'].startLaunch).toHaveBeenCalledWith(startLaunchObj);
+    });
+    test('reporter.launchId should be set', () => {
```
Comment (KatsiarynaTatarynovich) [inline src/__tests__/reporter/reporter.spec.ts:72]: You should write your code consistently. If you add a space (line 51), then add it everywhere before tests. In general, it is better to add spaces for better readability.

File: src/__tests__/reporter/reporter.spec.ts [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  });
+
+  return reporter;
+};
+const config = getDefaultMockConfig();
+
+describe('setup reporter', () => {
+  const reporter = setupReporter();
+  test('client instance should exist', () => {
+    expect(reporter['client']).toBeDefined();
```
Comment (KatsiarynaTatarynovich) [inline src/__tests__/reporter/reporter.spec.ts:46]: You can just write `reporter.client`

File: src/__tests__/reporter/reporter.spec.ts [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    expect(reporter['client']).toBeInstanceOf(RPClientMock);
+  });
+
+  test('config should be setup', () => {
+    expect(reporter['config']).toBeDefined();
+    expect(reporter['config']).toEqual(config);
```
Comment (KatsiarynaTatarynovich) [inline src/__tests__/reporter/reporter.spec.ts:52]: You can just write reporter.config

**Review by KatsiarynaTatarynovich — CHANGES_REQUESTED** [review]

File: src/__tests__/reporter/reporter.spec.ts (hunk 4 above)
Comment (chivekrodis) [reply src/__tests__/reporter/reporter.spec.ts:46]: it is just workaround to escape error like Property 'client' is private and only accessible within class 'Reporter'

File: src/__tests__/reporter/reporter.spec.ts (hunk 1 above)
Comment (chivekrodis) [reply src/__tests__/reporter/reporter.spec.ts]: my bad, unnecessary now

File: src/__tests__/reporter/reporter.spec.ts [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+      test('reporter.testItems should be []', () => {
+        expect(reporter['testItems']).toEqual([]);
+      });
+    });
+
+    describe('test with FAILED status', () => {
+      const reporter = setupReporter(['launchId', 'suiteIds', 'testItems']);
+      testRunInfo.skipped = false;
+      testRunInfo.errs = ['401 unauthorized'];
+      reporter.sendLogsOnFail = jest.fn((errors, testItemId) => ({ errors, testItemId }));
```
Comment (AmsterGet) [inline src/__tests__/reporter/reporter.spec.ts]: Why not just mock this function implementation?
F.e.
```javascript
jest.spyOn(reporter, 'sendLogsOnFail').mockImplementation(() => {});
```
It will simplify parameters cheching for [these](https://github.com/reportportal/agent-js-testcafe/pull/6/files#diff-0571ce57a1ca3a45fea6b987d7592d0a5712399e203134d2f2c19df3ee66013eR184-R188) lines.

**Review by AmsterGet — APPROVED** [review]
